x-pack/plugins/triggers_actions_ui/public/application/sections/alert_form/alert_add.tsx

```diff
@@ -121,16 +125,18 @@ export const AlertAdd = ({
 
   async function onSaveAlert(): Promise<Alert | undefined> {
     try {
-      const newAlert = await createAlert({ http, alert });
-      toastNotifications.addSuccess(
-        i18n.translate('xpack.triggersActionsUI.sections.alertAdd.saveSuccessNotificationText', {
-          defaultMessage: 'Created alert "{alertName}"',
-          values: {
-            alertName: newAlert.name,
-          },
-        })
-      );
-      return newAlert;
+      if (isValidAlert(alert, errors)) {
```

Comment:
I noticed the addition of `isValidAlert` check here. Does it mean there's a scenario the save code can get executed while the alert hasErrors (disabled save button) or just for safety?

Reply:
Type wise, yes, but not UX wise.
If we break these components down we might be able to improve that.

Reply:
Got it, thanks 👍